myapp/app/controllers/tasks_controller.rb

```diff
@@ -26,19 +26,21 @@ def index
     # TODO: support ascending
     sort += ' DESC'
 
-    @tasks = q.order(sort).page(params[:page])
+    @tasks = q.where(user_id: current_user.id).order(sort).page(params[:page])
 
     # for new form
     @new_task = Task.new
   end
 
   def show
     @task = Task.find_by(id: params[:id])
-    redirect_to error_path(404) if @task.nil?
+    return redirect_to error_path(404) if @task.nil?
+    return redirect_to error_path(401) if @task["user_id"] != current_user.id
```

Comment:
To access the attributes of an ActiveRecord object, it is recommended to use methods. This approach is generally preferred.
```suggestion
    return redirect_to error_path(401) if @task.user_id != current_user.id
```

Reply:
Addressed, Thank for the suggestion.